packages/generator/src/entity/class.ts

```diff
@@ -114,6 +142,122 @@ function property(prop: VdmProperty): PropertyDeclarationStructure {
   };
 }
 
+function boundFunction(
+  fn: VdmFunctionImport,
+  entity: VdmEntity,
+  service: VdmServiceMetadata
+): MethodDeclarationStructure {
+  return {
+    kind: StructureKind.Method,
+    name: fn.name,
+    returnType: `BoundFunctionRequestBuilder<${entity.className}<DeSerializersT>, DeSerializersT, any, ${fn.returnType.returnType} | null>`,
+    typeParameters: [
+      {
+        name: 'DeSerializersT extends DeSerializers = DefaultDeSerializers'
+      }
+    ],
+    parameters: boundFunctionsParameter(fn),
+    statements: boundFunctionsStatements(fn, entity, service)
+  };
+}
+
+function boundFunctionsParameter(
+  fn: VdmFunctionImport
+): ParameterDeclarationStructure[] {
+  if (!fn.parameters) {
+    return [];
+  }
+  return fn.parameters.map(p => ({
+    name: p.parameterName,
+    type: p.jsType,
+    kind: StructureKind.Parameter
+  }));
+}
+
+function boundFunctionsStatements(
+  fn: VdmFunctionImport,
+  entity: VdmEntity,
+  service: VdmServiceMetadata
+): string[] {
+  const name = `${service.namespaces[0]}.${fn.originalName}`;
+  const statements: string[] = boundFunctionsParameterStatements(fn).concat([
+    'const deSerializers = defaultDeSerializers as any;',
+    'return new BoundFunctionRequestBuilder(',
+    // fixme: do we need to do anything in the transformer function?
+    `this._entityApi as any, this as any, '${name}', (data) => data, params, deSerializers`,
+    ') as any;'
+  ]);
+  return statements;
+}
+
+function boundFunctionsParameterStatements(fn: VdmFunctionImport): string[] {
+  return ['const params = {'].concat(
+    fn.parameters?.map(
+      p =>
+        `${p.parameterName}: new FunctionImportParameter('${p.parameterName}', '${p.edmType}', ${p.parameterName}),`
+    ),
+    ['};']
+  );
```

Comment:
[req] I find this very odd to read. I think it would be better if this was either one string instead of an array of strings or if this function would just return the list of parameter statements.
Also, please use `array.join(,)` to concatenate the elements of the array with a comma.